src/lib/dnssd/Discovery_ImplPlatform.cpp

```diff
@@ -43,7 +43,7 @@ static void HandleNodeResolve(void * context, DnssdService * result, const Span<
 {
     DiscoveryContext * discoveryContext = static_cast<DiscoveryContext *>(context);
 
-    if (error != CHIP_NO_ERROR)
+    if (error != CHIP_NO_ERROR && error != CHIP_ERROR_IN_PROGRESS)
```

Comment:
As a followup, it really would be better for this to be an explicit argument to the callback.  Othewise it's too easy for someone to accidentally end up with CHIP_ERROR_IN_PROGRESS somewhere, especially since this behavior is completely undocumented, and get unexpected behavior.

Reply:
[Agreed.](https://github.com/project-chip/connectedhomeip/issues/35823)